### ISSUE TITLE
MaximumWidthRequest actually set WidthRequest.

### ISSUE THREAD

**Issue by KSemenenko** [dotnet/maui#11725]
### Description

I want my StackLayout to expand but no more than 450px.
But this property actually sets the width.
```
<StackLayout Margin="24,0" Spacing="16" HorizontalOptions="Center" MaximumWidthRequest="450" >
    <BoxView HeightRequest="24" Color="{StaticResource BackgroundMain}"/>	 
    <Button Style="{StaticResource SecondaryDefaultBig}" Text="{x:Static Localizations:AppResources.CANCEL}" Clicked="Cancel_Clicked" />
</StackLayout>
```

### Steps to Reproduce

Set MaximumWidthRequest to layout wich use Margin and stretching.

https://github.com/dotnet/maui/files/10117140/WidthIssue.zip

### Link to public reproduction project repository

no

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

iOS/android

### Did you find any workaround?

On the one hand a logical name, but on the other hand how do I set the maximum width of the component?

**Comment by hartez** [comment]
If you remove the BoxView, does it work as expected?

**Comment by KSemenenko** [comment]
I did repo for it.
so in original case seems it's ok, it depend on conent but for **HorizontalOptions="Fill"** it's not working.
But as for me it have to work, becase  maybe I would like to limit Fill with MaximumWidthRequest

[WidthIssue.zip](https://github.com/dotnet/maui/files/10117140/WidthIssue.zip)

<img width="913" alt="Screenshot 2022-11-29 at 21 39 32" src="https://user-images.githubusercontent.com/4385716/204643397-055ff4e1-ba8c-4f5d-9120-bf6a4df95b1a.png">

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 1.0. Repro on Android 13.0-API33 and  iOS 16.4 with provided Project:
[WidthIssue.zip](https://github.com/dotnet/maui/files/11684972/WidthIssue.zip)

![image](https://github.com/dotnet/maui/assets/97647558/b52b321c-3bc2-4734-87d7-c291adefccf7)

**Comment by hartez** [comment]
The issue with "Fill" was fixed by #15022.
